File: frontend/types/alerts.ts
/**
 * Types for player alerts and notifications system.
 */

export type AlertLevel = 'info' | 'warning' | 'critical';

export type PlayerAlert = {
  id: string;
  player_id: string;
  date: string;
  type: string;
  level: AlertLevel;
  message: string;
  resolved: boolean;
  created_at: string;
};

export type PlayerAlertsList = {
  player_id: string;
  alerts: PlayerAlert[];
};

export type TodayAlert = PlayerAlert & {
  player_name: string;
  jersey_number: number | null;
  role: string;
};

export type TodayAlertsList = {
  date: string;
  alerts: TodayAlert[];
};

export type ResolveAlertResponse = {
  status: string;
  message: string;
};

File: backend/app/schemas/alerts.py
"""Schemas for player alerts."""

from datetime import date, datetime
from typing import List, Literal
from uuid import UUID

from pydantic import BaseModel


class PlayerAlert(BaseModel):
    """Individual player alert."""
    id: str
    player_id: str
    date: date
    type: str
    level: Literal['info', 'warning', 'critical']
    message: str
    resolved: bool
    created_at: datetime

    class Config:
        from_attributes = True


class PlayerAlertsList(BaseModel):
    """List of alerts for a player."""
    player_id: UUID
    alerts: List[PlayerAlert]

    class Config:
        from_attributes = True


class TodayAlertsList(BaseModel):
    """List of all today's alerts across all players."""
    date: date
    alerts: List[dict]  # Contains player info + alert data

    class Config:
        from_attributes = True
